### PR TITLE
Retry PR

### PR DESCRIPTION
**Pull request by mymjack**
Oh those reverts? Yeah I screwed up pretty well. Had some conflicts but now resolved

### CODE REVIEW

File: findball.m [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-				 mask(round(y),round(x), f) = 1;
-            end         
-        end
-    end
-    % Dim original clip to display interest points
-    preview = 0.3*255*uint8(dclip);  
-    % Apply mask
-    preview(mask==1)=255;
-end
+function [xyif, preview] = findBall(dclip)
```
Comment (coder3000) [inline findball.m:1]: Can you change the filename to findBall?

File: fitcurve.m [hunk 2]
```diff
@@ -0,0 +1,25 @@
+function curve = fitcurve(segments)
```
Comment (coder3000) [inline fitcurve.m:1]: Not sure why we still have this.. Thought we gonna have one unified function?

File: nneighbours.m [hunk 3]
```diff
@@ -1,36 +1,84 @@
...
+        c(i, 5) = group;
+        c(targets(idx), 5) = group;
+        
+        notfound = 0;
+        while notfound < maxguess
+            
+            % Fit curve to current group
+            ingroup = cout(cout(:,5)==group, :);
+            framerange = sort(ingroup(:, 4));
+            if size(ingroup, 1) <=2
```
Comment (coder3000) [inline nneighbours.m:39]: Isn't px always linear and py quadratic?

File: nneighbours.m [hunk 4]
```diff
@@ -1,36 +1,84 @@
...
+                px = polyfit(ingroup(:,4),ingroup(:,1),1);
+                py = polyfit(ingroup(:,4),ingroup(:,2),1);
+            else
+                px = polyfit(ingroup(:,4),ingroup(:,1),2);
+                py = polyfit(ingroup(:,4),ingroup(:,2),2);
+            end
+            psx = polyfit(ingroup(:,4),ingroup(:,1),1);
```
Comment (coder3000) [inline nneighbours.m:46]: Why calculate this again?

File: nneighbours.m [hunk 5]
```diff
@@ -1,36 +1,84 @@
...
+        if cp(5) ~= 0  % Inner node of some curve is unlikely another start
+            continue;
+        end
+        
+        % Find first segment by linking to the closest point in next frame.
+        targets = find(c(:,4)==cp(4)+1);
+        [idx, d] = nearest(cp, c(targets,1:2));
+        if d > maxdist
+            targets = find(c(:,4)==cp(4)+2);
+            [idx, d] = nearest(cp, c(targets,1:2));
```
Comment (coder3000) [inline nneighbours.m:23]: You need find multiple second points that are within fixed distance

File: nneighbours.m [hunk 6]
```diff
@@ -1,36 +1,84 @@
...
-                    c(tidx, 5) = cp(5);
-                else
-                    c(p, 5) = group;
-                    c(tidx, 5) = group;
-                    group = group +1;
-                end
-                break; % We want the nearest one so stop
+    for i=1:size(c,1)
+        cp = c(i, :); 
+        if cp(5) ~= 0  % Inner node of some curve is unlikely another start
```
Comment (coder3000) [inline nneighbours.m:14]: Clever, but what if a point in second frame is closest to two points in first frame simultaneously?

File: processClip.m [hunk 7]
```diff
@@ -19,30 +19,29 @@
...
 % Noise suppression
-clip = imfilter(clip, g);
+clip = imfilter(clip0, g);
 % Get binary frame difference
 dclip = diff3(clip);
-% IDEA: detect and cut camera movement
-dclip = detectCameraMov(dclip);
 % Filter ball candidates
-[candidates, cclip] = findBall(dclip);
+[candidates, ~] = findBall(dclip);
```
Comment (coder3000) [inline processClip.m:32]: We may need to retain cclip for presentation

File: nneighbours.m [hunk 8]
```diff
@@ -1,36 +1,84 @@
-function c = nneighbours(c, clip)
-    % Image origin is top left. We want bottom left so invert y
-    x = figure;plot(c(:, 4), c(:, 1), 'k.'); hold on;
-    y = figure;plot(c(:, 4), size(clip,1)-c(:, 2), 'k.'); hold on;
-    c = [c zeros(size(c,1), 1)];
-    group = 1; % Connected points share a same group id.
-    maxdist = 2600; % Max neighbour distance between frames.
+function cout = nneighbours(candidates)
```
Comment (coder3000) [inline nneighbours.m:1]: Where are coefficients of px and py saved?

**Review by coder3000 — CHANGES_REQUESTED** [review]
Still has some problems

File: nneighbours.m (hunk 3 above)
Comment (mymjack) [reply nneighbours.m:39]: It was, still can be

File: nneighbours.m (hunk 4 above)
Comment (mymjack) [reply nneighbours.m:46]: Straightness penalize purpose

File: nneighbours.m (hunk 6 above)
Comment (mymjack) [reply nneighbours.m:14]: Then the first two points in first frame should both find this point in second frame. The challenge is the reverse.

**Pull request closed without merging**